torch/utils/_pytree.py

```diff
@@ -1149,17 +1149,17 @@ def tree_map_(
 # These specializations help with type inference on the lambda passed to this
 # function
 @overload
-def map_only(type_or_types_or_pred: Type2[T, S], /) -> MapOnlyFn[Fn2[T, S, Any]]:
+def map_only(type_or_types_or_pred: type[T], /) -> MapOnlyFn[Fn[T, Any]]:
```

Comment:
Also slight nit that can be addressed in a latter PR. I think all these Type2, Type3, etc TypeVars/Aliases can be replaced by TypeVarTuple, right?

Reply:
We can replace `TypeN[...]` with `TypeVarTuple`. However, for `FnN[..., R]`, it does not allow:

```python
if sys.version_info >= (3, 10):
    Type1 = TypeVar("Type1", bound=type)
else:
    Type1 = TypeVar("Type1", bound=(type, types.UnionType))
Types = TypeVarTuple("Types", default=Type1)

Fn1 = Callable[[Type1], R]
Fns = Callable[[Union[Unpack[Types]]], R]
                      ^------------^: TypeVarTuple is not allowed in this context
```